frontend/src/metabase/query_builder/components/filters/pickers/BooleanPicker/BooleanPicker.tsx

```diff
@@ -21,29 +27,50 @@ const EXPANDED_OPTIONS: RadioOption<string | boolean | any>[] = [
 ];
 
 interface BooleanPickerProps {
-  className?: string;
   filter: Filter;
   onFilterChange: (filter: Filter) => void;
+  className?: string;
+  asCheckBox?: boolean;
 }
 
 function BooleanPicker({
   className,
   filter,
   onFilterChange,
+  asCheckBox = false,
```

Comment:
I hear you on the flag props, but there is some filter logic here that I wanted to avoid duplicating in a new component - and I think the addition is mostly display based. Let me think if I can come up with a way to have our cake and eat it too.